### ISSUE TITLE
XMonad config file is no longer optional

### ISSUE THREAD

**Issue by MasseR** [nix-community/home-manager#1891]
### XMonad config file is no longer optional

```
error: cannot coerce null to a string, at /nix/var/nix/profiles/per-user/root/channels/home-manager/modules/services/window-managers/xmonad.nix:105:11
```

The change c1761366b522595ff0dda47d7ba79d5242ecb31e introduced a way to compile the xmonad configuration during nix switch time. The script managing this copies the defined config file into xmonad.hs, but since the config file is nullable, this will fail when config is set to null (the default).

```
cp ${cfg.config} xmonad-config/xmonad.hs
```

#### Maintainer CC

@markusscherer 

#### Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.107, NixOS, 21.05pre279012.ad47284f8b0 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"home-manager, nixos-21.05pre279012.ad47284f8b0"`
 - channels(masse): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by MasseR** [comment]
Somewhat related, more of a warning. If you have a xmonad configuration that includes a binding for both recompiling and restarting xmonad, it will override the nix managed xmonad symlink with the new binary. This will cause later nix rebuilds to fail because the file already exists.

**Comment by cassandracomar** [comment]
this is concerning -- my xmonad config is close to 1k lines and I don't want to load it through my nix configuration. I would much prefer to let xmonad manage itself and to continue managing its configuration as previously. can this functionality become toggleable?

**Comment by berbiche** [comment]
A fix is available in #1893, to test you can update your Home Manager channel or flake input to the pull request:

- channel: `nix-channel --add https://github.com/nix-community/home-manager/archive/pull/1893/head.tar.gz`
- flake: update your `inputs.home-manager` (or whatever name you use) to `inputs.home-manager = "github:nix-community/home-manager/pull/1893/head"`
